### PR TITLE
add a checkbox to path edit form to allow ignore

### PR DESCRIPTION
**Pull request by kaligrafy**
default dwell times for associated nodes, and add help to default dwell time

### CODE REVIEW

**Comment by tahini** [comment]
The test failure looks legit

**Comment by tahini** [comment]
The ignore checkbox does not seem to have any effect...

My nodes have a dwell time of 20 seconds, I put 10, and whether I ignore or not, the stats are the same. If I change the dwell time of the nodes to 10, then the stats do decrease, so it's really the ignore checkbox not working.

File: packages/transition-common/src/services/path/Path.ts [hunk 1]
```diff
@@ -991,6 +992,24 @@ export class Path extends MapObject<GeoJSON.LineString, PathAttributes> implemen
         }
     }
 
+    getDwellTimeSecondsAtNode(nodeDwellTimeSeconds: number | undefined) : number {
+        const defaultGeneralDwellTimeSeconds = Math.max(0, Preferences.get('transit.nodes.defaultDwellTimeSeconds', 20));
+        const pathDwellTimeSeconds : number = Math.max(0,this.getData('defaultDwellTimeSeconds', defaultGeneralDwellTimeSeconds) as number);
+        if (this.getData('ignoreNodesDefaultDwellTimeSecond', false) === true) {
```
Comment (tahini) [inline packages/transition-common/src/services/path/Path.ts:998]: with an 's'

**Review by tahini — CHANGES_REQUESTED** [review]
Typo on the data field name, but when fixed, it works well

File: packages/transition-common/src/services/path/__tests__/Path.test.ts [hunk 2]
```diff
@@ -305,3 +305,53 @@ test('getClonedAttributes', () => {
...
+    });
+
+    it('should return the default general dwell time when node dwell time is undefined', () => {
+        instance.attributes.data.ignoreNodesDefaultDwellTimeSecond = false;
+        mockPreferencesGet.mockReturnValueOnce(20); // Default general dwell time
+        expect(instance.getDwellTimeSecondsAtNode(undefined)).toBe(20);
+    });
+
+    it('should return the path dwell time when ignoreNodesDefaultDwellTimeSecond is true', () => {
+        instance.attributes.data.ignoreNodesDefaultDwellTimeSecond = true;
```
Comment (tahini) [inline packages/transition-common/src/services/path/__tests__/Path.test.ts:334]: Typo in the unit tests as well

Comment (kaligrafy) [reply packages/transition-common/src/services/path/__tests__/Path.test.ts:334]: fixed

File: packages/transition-common/src/services/path/Path.ts [hunk 3]
```diff
@@ -991,6 +992,28 @@ export class Path extends MapObject<GeoJSON.LineString, PathAttributes> implemen
         }
     }
 
+    getDwellTimeSecondsAtNode(nodeDwellTimeSeconds: number | undefined) : number {
+        const defaultGeneralDwellTimeSeconds = Math.max(0, Preferences.get('transit.nodes.defaultDwellTimeSeconds', 20));
+        const pathDwellTimeSeconds : number = Math.max(0,this.getData('defaultDwellTimeSeconds', defaultGeneralDwellTimeSeconds) as number);
+        console.log('defaultGeneralDwellTimeSeconds', defaultGeneralDwellTimeSeconds);
+        console.log('pathDwellTimeSeconds', pathDwellTimeSeconds);
+        console.log('ignoreNodesDefaultDwellTimeSeconds', this.getData('ignoreNodesDefaultDwellTimeSeconds'));
```
Comment (tahini) [inline packages/transition-common/src/services/path/Path.ts:1000]: Please remove

**Review by tahini — CHANGES_REQUESTED** [review]

File: packages/transition-common/src/services/path/Path.ts (hunk 3 above)
Comment (kaligrafy) [reply packages/transition-common/src/services/path/Path.ts]: fixed

File: packages/transition-common/src/services/path/__tests__/PathData.test.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export const arbitraryData = {
     defaultAcceleration: 1.1,
     defaultDeceleration: 1.1,
     defaultDwellTimeSeconds: 30,
+    ignoreNodesDefaultDwellTimeSecond: false,
```
Comment (tahini) [inline packages/transition-common/src/services/path/__tests__/PathData.test.ts:23]: here, missing an 's'

**Review by tahini — CHANGES_REQUESTED** [review]
You probably should do a find & replace on the whole code to see if any other typos are lying around...

**Review by tahini — APPROVED** [review]

**Pull request merged**